src/constructs/aws/DatabaseSql.ts

```diff
@@ -0,0 +1,161 @@
+            minimum: 20,
+        },
+    },
+    additionalProperties: false,
+    required: [],
+} as const;
+
+type Configuration = FromSchema<typeof SCHEMA>;
+
+export class DatabaseSql extends AwsConstruct {
+    public static type = "database/sql";
+    public static schema = SCHEMA;
+
+    private readonly dbInstance: DatabaseInstance;
+    private readonly dbHostOutput: CfnOutput;
+    private readonly username: string;
+    private passwordSecretName: string | undefined;
+
+    constructor(
+        scope: CdkConstruct,
+        private readonly id: string,
+        private readonly configuration: Configuration,
+        private readonly provider: AwsProvider
+    ) {
+        super(scope, id);
+
+        const vpc = provider.enableVpc();
+
+        this.username = this.configuration.username ?? "admin";
+
+        this.dbInstance = new DatabaseInstance(this, "Instance", {
+            // https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseInstance.html#construct-props
+            instanceIdentifier: configuration.name ?? `${this.provider.stackName}-${id}`,
+            databaseName: this.safeDbName(configuration.name ?? `${this.provider.stackName}-${id}`),
+            engine: this.getEngineVersion(),
+            instanceType: new InstanceType(configuration.instanceType ?? "t3.micro"),
+            credentials: this.credentials(),
+            vpc,
+            // Put the instance in the private subnet
+            vpcSubnets: {
+                subnetType: SubnetType.PRIVATE,
+            },
+            // We go with 20 (the minimum) instead of 100 by default, because it is much cheaper
+            allocatedStorage: configuration.storageSize ?? 20,
+            // 7 days backups instead of the default 1
+            backupRetention: Duration.days(7),
+            cloudwatchLogsExports: this.logsToEnable(),
+        });
+
+        this.dbHostOutput = new CfnOutput(this, "DbHost", {
+            value: this.dbInstance.instanceEndpoint.hostname,
+        });
+    }
+
+    private credentials() {
+        const password = this.configuration.password ?? "";
+        if (password === "") {
+            // If no password is defined, a random one will be generated and stored in Secrets Manager
+            this.passwordSecretName = `${this.provider.stackName}/${this.id}/password`;
+
+            return Credentials.fromGeneratedSecret(this.username, {
+                secretName: this.passwordSecretName,
+            });
+        }
+
+        return Credentials.fromPassword(this.username, SecretValue.plainText(password));
+    }
+
+    commands(): Record<string, () => void | Promise<void>> {
+        return {};
+    }
+
+    outputs(): Record<string, () => Promise<string | undefined>> {
+        return {
+            host: () => this.provider.getStackOutput(this.dbHostOutput),
+        };
+    }
+
+    variables(): Record<string, unknown> {
+        return {
+            host: this.dbInstance.instanceEndpoint.hostname,
```

Comment:
It would be nice to construct a connection string because lots of frameworks/libraries encourage to use them by default (something like `mysql://user:pass@host:port/dbName`)

I searched a bit and the format seems consistent across libraries. The only exception is Java JDBC which is mostly the same but prefixed with `jdbc:`

Reply:
Agreed, it wouldn't cost much I think.